libs/cardano-ledger-core/src/Cardano/Ledger/TxIn.hs

```diff
@@ -72,101 +71,59 @@ newtype TxId crypto = TxId {_unTxId :: SafeHash crypto EraIndependentTxBody}
   deriving (Show, Eq, Ord, Generic)
   deriving newtype (NoThunks, HeapWords)
 
+deriving newtype instance CC.Crypto crypto => Keyed (TxId crypto)
+
 deriving newtype instance CC.Crypto crypto => ToCBOR (TxId crypto)
 
 deriving newtype instance CC.Crypto crypto => FromCBOR (TxId crypto)
 
 deriving newtype instance CC.Crypto crypto => NFData (TxId crypto)
 
-instance HeapWords (TxIn crypto) where
-  heapWords (TxInCompact32 a _ _ _ ix) =
-    6 + (4 * HW.heapWordsUnpacked a) + HW.heapWordsUnpacked ix
-  heapWords (TxInCompactOther tid ix) =
-    3 + HW.heapWords tid + HW.heapWordsUnpacked ix
+instance CC.Crypto crypto => HeapWords (TxIn crypto) where
+  heapWords (TxIn txId txIx) =
+    2 + HW.heapWords txId + HW.heapWordsUnpacked txIx
 
 -- | The input of a UTxO.
-data TxIn crypto where
-  TxInCompact32 ::
-    HS.SizeHash (CC.HASH crypto) ~ 32 =>
-    {-# UNPACK #-} !Word64 -> -- Hash part 1/4
-    {-# UNPACK #-} !Word64 -> -- Hash part 2/4
-    {-# UNPACK #-} !Word64 -> -- Hash part 3/4
-    {-# UNPACK #-} !Word64 -> -- Hash part 4/4
-    {-# UNPACK #-} !Word64 -> -- Index
-    TxIn crypto
-  TxInCompactOther :: !(TxId crypto) -> {-# UNPACK #-} !Word64 -> TxIn crypto
+data TxIn crypto = TxInCompact !(TxId crypto) {-# UNPACK #-} !Int
```

Comment:
Why the change to `Int` here?

Reply:
Because we will use it with `IntMap`, which needs an `Int`, rather than a `Word64`. I don't think it would a problem, right? Considering that max value on mainnet is 249 and I don't see tx index going larger than `maxBound :: Int` any time in the foreseeable future :smile: